jsonnet/rules.jsonnet

```diff
@@ -206,6 +206,31 @@ local droppedKsmLabels = 'endpoint, instance, job, pod, service';
             expr: 'sum(cluster:master_nodes or on(node) kube_node_labels ) BY (label_beta_kubernetes_io_instance_type, label_node_role_kubernetes_io, label_kubernetes_io_arch, label_node_openshift_io_os_id)',
             record: 'cluster:node_instance_type_count:sum',
           },
+          {
+            // Track the number of physical cores that are considered accessible for general workloads to run on. A physical core is an unshared CPU as seen by the node operating system, ignoring hyperthreading or virtualization. The sum of all non-infrastructure node physical cores plus master node physical cores (if masters are schedulable) is considered available for workload use.
+            expr: 'sum(node_role_os_version_machine:cpu_capacity_cores:sum{label_node_role_kubernetes_io_master="",label_node_role_kubernetes_io_infra=""} or absent(node_role_os_version_machine:cpu_capacity_cores:sum{label_node_role_kubernetes_io_master="",label_node_role_kubernetes_io_infra=""})*0) + (sum(node_role_os_version_machine:cpu_capacity_cores:sum{label_node_role_kubernetes_io_master="true"}) * ((max(cluster_master_schedulable == 1)*0+1) or (absent(cluster_master_schedulable == 1)*0)))',
```

Comment:
OSD uses `infra` for compute-side OCP-core components (router, metrics, etc.).  But for folks running OCP directly, whether they have infra nodes or the semantics of `infra` are up to individual cluster admins, right?  E.g. see one setup which allowed user workloads on the `infra` nodes in [rhbz#1760675](https://bugzilla.redhat.com/show_bug.cgi?id=1760675).  It's not clear to me that we can blanket-exclude `infra` cores from user workload counts.

Reply:
If a customer gives a node the "infra" role, they are explicitly telling us they don't want to be paid for it.  So we do blanket, and it's up to the customer for that to track as per our contract

Reply:
just to clarify:

will this map to physical capacity on workers and not physical allocatable capacity on workers?  meaning we do bill on any reserved cpu given to operation of the node itself.

Reply:
This maps today to the machine capabilities, not to the schedulable capacity.  If in future we decide to change that, we would probably do so by altering this metric (this metric is recording the workload physical cores).  We bill on reserved yes.